pkg/apis/testing/v1alpha1/testsuite_types.go

```diff
@@ -0,0 +1,138 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// TestSuiteSpec defines the desired state of TestSuite
+type TestSuiteSpec struct {
+	// How many tests we want to execute at the same time.
+	// Depends on cluster size and it's load
+	ConcurrencyLevel int64 `json:"concurrencyLevel,omitempty"`
+	// You can specify test names explicitly.
+	TestNamesSelector []NamespacedTest `json:"testNamesSelector,omitempty"`
+	// Will run every test that depends of any of the component listed here
+	ComponentsSelector []string `json:"componentsSelector,omitempty"`
+	// Running all tests from suite cannot take more time that specified here
+	SuiteTimeout *metav1.Duration `json:"suiteTimeout,omitempty"`
+	// If specific TestDefinition does not define timeout, use this one
+	DefaultTestTimeout *metav1.Duration `json:"defaultTestTimeout,omitempty"`
+	// Should I repeat every test? Default value will be 1
+	Repeat int64 `json:"repeat,omitempty"`
```

Comment:
shouldn't this field by specified on `TestDefinitionSpec` level? 

scenario with Repeat on TestSuiteSpec:
1. Adding new tests
2. Define the TestDefinition
3. Change the Repeat in TestSuiteSpec 
4. Narrow TestNamesSelector in TestSuiteSpec to run the only test that you are interested in

scenario with Repeat on TestDefinitionSpec:
1. Adding new tests
2. Define the TestDefinition with proper Repeat

Reply:
Thanks to having Repeat on suite level you can sue same scenario to implement functional tests and stress tests only by creating new suite.

Reply:
The question is how can I specify repeat per TestDefinition if they differ?

Reply:
I disagree. In most cases, if you want to repeat some test, you don't want to run other tests - because you are focusing on one unstable test. Because of that, you should define new TestSuite to run an only specific test. It can sound a little bit complicated, but remember that it will be simplified by good CLI.
So far we have the following fields in the test definition:
- skip - 
- disableConcurrency

In both cases an author of the test can provide reasonable values: he knows if that test can be executed with other tests concurrently or if a test is ready and can be executed. How many times you want to repeat a given test, it is rather a question to a person who execute tests and who define TestSuite.